components/datetime/src/neo_skeleton.rs

```diff
@@ -651,6 +654,21 @@ impl NeoTimeComponents {
         }
     }
 
+    pub(crate) fn with_hour_cycle(self, hour_cycle: &Value) -> Self {
+        const H11: Subtag = icu_locale_core::subtags::subtag!("h11");
+        const H12: Subtag = icu_locale_core::subtags::subtag!("h12");
+        let explicit_h12 = matches!(hour_cycle.as_single_subtag(), Some(&H11) | Some(&H12));
+        match (self, explicit_h12) {
+            (Self::Hour, true) => Self::Hour12,
```

Comment:
nit: I would find this more readable
```suggestion
        match self {
            Self::Hour if explicit_h12 => Self::Hour12,
```